### PR TITLE
RN v0.60 upgrade immediate-term followup

### PR DESCRIPTION
**Pull request by chrisbobbe**
This doesn't include as much as I was hoping to include; see details at https://github.com/zulip/zulip-mobile/issues/3548#issue-465925049.

In particular (these are the major things), it doesn't include using Hermes (#4131) or upgrading Unimodules (#4091). But Hermes, IIUC, can be done any time; the Unimodules upgrade should Just Work, but only following the RN v0.61 upgrade.

Fixes: #4118

### CODE REVIEW

**Comment by gnprice** [comment]
Second commit message has a `[1]` but there's no footnote -- I'm curious to follow that link :slightly_smiling_face:

**Comment by chrisbobbe** [comment]
Thanks for catching that! I'll add that in—along with a note about how the "default screen orientation" bit, in those instructions, is misleading (see https://github.com/zulip/zulip-mobile/issues/4118#issuecomment-641561847).

**Comment by gnprice** [comment]
Mentioning the issue one of these fixes, so it gets a backlink to the PR: this fixes #4118.

**Comment by chrisbobbe** [comment]
> Mentioning the issue one of these fixes, so it gets a backlink to the PR: this fixes #4118.

I was just about to do that, when I saw you posted. Thanks!

File: src/boot/AppEventHandlers.js [hunk 1]
```diff
@@ -69,13 +69,30 @@ type Props = $ReadOnly<{|
...
   netInfoDisconnectCallback: (() => void) | null = null;
 
-  handleOrientationChange = (orientation: OrientationT) => {
+  handleOrientationChange = (event: ScreenOrientation.OrientationChangeEvent) => {
     const { dispatch } = this.props;
-    dispatch(appOrientation(orientation));
+
+    const {
+      orientationInfo: { orientation },
+    } = event;
```
Comment (gnprice) [inline src/boot/AppEventHandlers.js:93]: The multi-line formatting is kind of a sign here of the fact that this syntax takes a bit  of effort for the reader to unpack :wink: 

The usual way we'd write this is:
```js
const { orientation } = event.orientationInfo;
```

**Comment by chrisbobbe** [comment]
Thanks for the comments! I've pushed a new revision.

**Comment by gnprice** [comment]
Looks good, thanks! I've added a few words to the commit message with your findings from https://github.com/zulip/zulip-mobile/issues/4118#issuecomment-641621448 ; merging.

**Pull request merged**